lib/ratchet/transformer.ex

```diff
@@ -10,52 +10,26 @@ defmodule Ratchet.Transformer do
   def transform([child|rest], scope) do
     [transform(child, scope)|transform(rest, scope)]
   end
-
   def transform(element, scope) do
-    case get_type(element) do
-      :none -> transform_element(element, scope)
-      type -> transform_element(type, element, scope)
-    end
-  end
-
-  @doc """
-  Get the "type" of a node. First property or scope is returned.
-
-      iex> Transformer.get_type({"div", [{"data-scope", "foo"}], []})
-      {:scope, "foo"}
-      iex> Transformer.get_type({"div", [{"data-prop", "bar"}], []})
-      {:property, "bar"}
-      iex> Transformer.get_type({"div", [{"data-scope", "foo"}, {"data-prop", "bar"}], []})
-      {:scope, "foo"}
-      iex> Transformer.get_type({"div", [], []})
-      :none
-  """
-  def get_type({_tag, attributes, _children}) do
-    Enum.find_value attributes, :none, fn
-      {"data-scope", scope} -> {:scope, scope}
-      {"data-prop", prop} -> {:property, prop}
-      _ -> false
-    end
+    transform_element(element, scope)
   end
 
-  defp transform_element({:scope, property}, {tag, attributes, children}, scope) do
-    children = transform(children, property) |> List.flatten
+  defp transform_element({tag, attributes = [{"data-scope", data_value}], children}, scope) do
```

Comment:
Unfortunately this won't work for lists. In Elixir, lists don't support partial matches, therefore this attribute would only match in the case of attributes containing exactly one element in the specified form. We'd want our elements to be able to contain _anything_ that _includes_ a scope or prop.